web/src/onboarding_steps.ts

```diff
@@ -21,8 +21,10 @@ export function post_onboarding_step_as_read(onboarding_step_name: string): void
     });
 }
 
-export function filter_new_hotspots(onboarding_steps: OnboardingSteps[]): OnboardingSteps[] {
-    return onboarding_steps.filter((onboarding_step) => onboarding_step.type === "hotspot");
+export function filter_new_hotspots(onboarding_steps: OnboardingSteps[]): RawHotspot[] {
+    return onboarding_steps.filter(
+        (onboarding_step): onboarding_step is Hotspot => onboarding_step.type === "hotspot",
+    );
```

Comment:
This is a user-defined type guard. These are not checked for correctness by TypeScript, so they should be avoided. Use `.flatMap` instead.

```suggestion
    return onboarding_steps.flatMap((onboarding_step) =>
        onboarding_step.type === "hotspot" ? [onboarding_step] : [],
    );
```